components/sidebar/sidebar_channel/sidebar_channel_menu/sidebar_channel_menu.tsx

```diff
@@ -209,16 +209,32 @@ export class SidebarChannelMenu extends React.PureComponent<Props, State> {
 
         let leaveChannel;
         if (channel.name !== Constants.DEFAULT_CHANNEL) {
-            leaveChannel = (
-                <Menu.Group>
-                    <Menu.ItemAction
-                        id={`leave-${channel.id}`}
-                        onClick={this.handleLeaveChannel}
-                        icon={<i className='icon-close'/>}
-                        text={leaveChannelText}
-                    />
-                </Menu.Group>
-            );
+            if (channel.type === Constants.DM_CHANNEL || channel.type === Constants.GM_CHANNEL) {
+                leaveChannel =
+                    (<Menu.Group>
+                        <Menu.ItemAction
+                            id={`leave-${channel.id}`}
+                            onClick={this.handleLeaveChannel}
+                            icon={<i className='icon-close'/>}
```

Comment:
we should be using the icon components from `@mattermost/compass-icons/components` here. It takes a `size` and `color` prop. `size` should be `16` and `color` should be `'currentColor'` so it does take the colro that is being set by the component for each state (default, hovering, active, etc.)

Reply:
could you add this change in as well please?